lib/src/metta/runner/environment.rs

```diff
@@ -16,16 +18,21 @@ use directories::ProjectDirs;
 #[derive(Debug)]
 pub struct Environment {
     config_dir: Option<PathBuf>,
+    caches_dir: Option<PathBuf>,
     init_metta_path: Option<PathBuf>,
     working_dir: Option<PathBuf>,
     is_test: bool,
     #[cfg(feature = "pkg_mgmt")]
     catalogs: Vec<Box<dyn ModuleCatalog>>,
     #[cfg(feature = "pkg_mgmt")]
     pub(crate) fs_mod_formats: Arc<Vec<Box<dyn FsModuleFormat>>>,
+    /// The store for modules loaded from git by explicit URL
+    #[cfg(feature = "pkg_mgmt")]
+    pub(crate) explicit_git_mods: Option<LocalCatalog>,
```

Comment:
I figured out the answer probably. All git catalogs are added into `catalogs` but when some git module is imported explicitly (without using a catalog) it is placed into `explicit_git_mods`. Now I have a question why it is different from non-git modules. If non-git modules are not placed into some catalog then why git modules cannot be the same. If non-git modules are always loaded from some catalog why git modules cannot be the same?

Reply:
My understanding from above is that main difference between `explicit_git_mods` and `catalogs` is that modules in `explicit_git_mods` should not be searchable by name as it happens with other catalogs. May be we can add an implementation of the `ModuleCatalog` which always returns empty `Vec` on lookup. Then for each git module added explicitly we could instantiate such catalog and add it into `catalogs` collection. Would it work?

Reply:
I don't understand why that would be better.  The main (only) purpose of the `Environment::catalogs` list is to express a search-priority order.  So I think adding an unsearchable catalog to that list defeats the purpose.

The single `explicit_git_mods` catalog is there to manage modules that are "installed" in a practical sense (on the user's machine, ready to load, able to be deleted, upgraded, etc.) but not part of the module search path.

Is your objection that you don't like that it's a special case for the purposes of listing and management?

Reply:
It is not an objection, I am asking this questions to better understand the purpose of the field. At first I saw the field with `git_` prefix in the structure which is not Git specific that sounds like abstraction leakage. Now my understanding this is a collection explicitly loaded modules not necessary Git-based. Thus may be it would be more clear if field is named like `explicitly_loaded_mods` without Git inside.

Reply:
That's a good point.  After refactoring the way this interacts with the GitCatalog underneath, it's no longer git-specific.  I'll rename the catalog.